src/schemas/resolvers/resolver_query.ts

```diff
@@ -1,29 +1,77 @@
-import {
-  getAccounts,
-  getProducts,
-  getAccount,
-  getProduct,
-} from "../functions/get";
+import { UserInputError } from "apollo-server-core";
+import { getAccount, getProduct } from "../functions/get";
+import { Product } from "../functions/producttypes";
+import { ID, User } from "../functions/usertypes";
 
 export const queries_resolver = {
   Query: {
-    accounts: async () => {
-      return getAccounts();
+    node: async (_: undefined, { id }: ID) => {
+      let result: any;
+      if (id.includes("product", 32)) {
+        result = await getProduct(id);
+        if (result) {
+          return {
+            __typename: "Product",
```

Comment:
You can resolve this via `Node.__resolveType` for example:
```ts
Node: {
  if (id.includes('product', 32)) return 'Product';
  else if ((id.includes('account', 32)) return 'Account';
  else return;
}
```

There's no need to return an error for this, we just tell the end-user that there's no data found.